ansible_runner/utils/__init__.py

```diff
@@ -413,12 +413,18 @@ def open_fifo_write(path, data):
     # If the data is a string instead of bytes, convert it before writing the fifo
     if isinstance(data, string_types):
         data = data.encode()
-    threading.Thread(target=lambda p, d: open(p, 'wb').write(d),
+
+    def worker(path, data):
+        with open(path, 'wb') as fh:
+            fh.write(data)
+        fh.close()
```

Comment:
Is `fh.close()` required outside of with context?

Reply:
Yep, that is what sorts the warning. I remember that i had to do this in other places to avoid runtime warnings as the release is not guaranteed w/o it.

Reply:
same here, I thought the context manager takes care of the close operation of the file handle.

Reply:
I can totally see a warning on the original `open().write()`, but if there's a warning with the context manager, I would find that very curious. Python docs are pretty clear that the context manager will guarantee the `close()` (https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files).

Reply:
I removed that line because I was not able to find where I read about it. We will find-out later if it was needed, when we include pylint as a pre-commit hook.